core/scripts/chaincli/handler/scrape_node_config.go

```diff
@@ -0,0 +1,270 @@
+	}
+
+	if !cmp.Equal(node.Ocr2ConfigPublicKey, ni.Ocr2ConfigPublicKey) {
+		diffs++
+		log.Errorf("OCR2 Config Public Key differs. The node returns %s but weiwatcher has %s", node.Ocr2ConfigPublicKey, ni.Ocr2ConfigPublicKey)
+	}
+
+	return diffs == 0
+}
+
+func (h *baseHandler) ScrapeNodes() {
+	log, closeLggr := logger.NewLogger()
+	logger.Sugared(log).ErrorIfFn(closeLggr, "Failed to close logger")
+
+	ctx := context.Background()
+	h.scrapeNodes(ctx, log)
+}
+
+func (h *baseHandler) scrapeNodes(ctx context.Context, log logger.Logger) {
+	log.Warn("This scrapes node address, peer ID, CSA node address, CSA public key, OCR2 ID, OCR2 config pub key, OCR2 onchain pub key, and OCR2 offchain pub key.")
+	log.Warn("This does NOT scrape for payee address, admin address etc. Please verify that manually.")
+	cls := make([]cmd.HTTPClient, len(h.cfg.KeeperURLs))
+	for i := range h.cfg.KeeperURLs {
+		url := h.cfg.KeeperURLs[i]
+		email := h.cfg.KeeperEmails[i]
+		if len(email) == 0 {
+			email = defaultChainlinkNodeLogin
+		}
+		pwd := h.cfg.KeeperPasswords[i]
+		if len(pwd) == 0 {
+			pwd = defaultChainlinkNodePassword
+		}
+
+		cl, err := authenticate(url, email, pwd, log)
+		if err != nil {
+			log.Fatal(err)
+		}
+		cls[i] = cl
+	}
+
+	nodes := map[string]*NodeInfo{}
+	var wg sync.WaitGroup
+	for i, cl := range cls {
+		wg.Add(1)
+		go h.scrapeNodeInfo(&wg, i, cl, nodes, log)
+	}
+	wg.Wait()
+
+	// if node info is not in RDD and weiwatchers, don't proceed further
+	if !h.cfg.VerifyNodes {
+		return
+	}
+	nodeInfos := h.fetchNodeInfos(ctx, log)
+	cnt := 0
+	for _, ni := range nodeInfos {
+		nodeAddr := ni.NodeAddress[0]
+		node := nodes[nodeAddr]
+		if node == nil {
+			continue
+		}
+		cnt++
+
+		log.Infof("start comparing data for node %s", nodeAddr)
+		if node.Equals(ni, log) {
+			log.Infof("node %s info is correct", nodeAddr)
+		} else {
+			log.Errorf("node %s info differs between the node instance and weiwatcher", nodeAddr)
+		}
+	}
+
+	if cnt != len(nodes) {
+		log.Infof("there are %d nodes provisioned , but .env is missing %d nodes", len(nodes), len(nodes)-cnt)
+	}
+}
+
+func (h *baseHandler) fetchNodeInfos(ctx context.Context, log logger.Logger) []NodeInfo {
+	client := http.DefaultClient
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, h.cfg.NodeConfigURL, nil)
+	if err != nil {
+		log.Fatalf("failed to build a GET request: %s", err)
+	}
+
+	resp, err := client.Do(req)
+	if err != nil {
+		log.Fatalf("failed to make a GET request: %s", err)
+	}
+	defer resp.Body.Close()
+
+	var nodeInfos []NodeInfo
+	if err := json.NewDecoder(resp.Body).Decode(&nodeInfos); err != nil {
+		log.Fatalf("failed to read response: %s", err)
+	}
+
+	return nodeInfos
+}
+
+func (h *baseHandler) scrapeNodeInfo(wg *sync.WaitGroup, i int, cl cmd.HTTPClient, nodes map[string]*NodeInfo, log logger.Logger) {
+	defer wg.Done()
+
+	// get node addresses
+	resp, err := nodeRequest(cl, "/v2/keys/eth")
```

Comment:
Can we extract the `/v2/keys/eth`, `/v2/keys/csa` and `/v2/keys/p2p` strings into constants? That way we can use the constant name to better describe what those URLs point to, e.g. `nodeAddressesEndpoint`, and then that would enable us to get rid of the comment above?

Reply:
done